src/lib/snark_worker_lib/rpcs.ml

```diff
@@ -104,7 +104,7 @@ module Make (Inputs : Intf.Inputs_intf) = struct
     module V1 = struct
       module T = struct
         type query =
-          ( ( Statement.Stable.V1.t
+          ( ( Statement.t
```

Comment:
I don't understand this change. Don't we want the `V1` in the RPC type to refer to only versioned modules?

Reply:
The module Statement can be different for each application of `RPC.Make`. As a result, I removed the Stable version of this module, like in Sparse ledger.

Reply:
I believe that `Rpc.Make` is called exactly once in the code, in `Worker.Make` (and I believe that functor is called once also, in either `Prod` or `Debug`). Therefore, inside `Rpc.Make`, it's OK to choose particular versions from the functor arguments. In particular, in `Inputs`, the module `Statement` is versioned, and`Statement.Stable.V1.t` should never change, which is needed in order for the RPC type to be versioned. On the other hand, `Statement.t` can change, which is what we want to avoid.

Does that sound right?

Reply:
If a functor is called once, it can choose versions from its inputs. If it can be called more than once, the caller has to provide the version. That's how I'm thinking ...

Reply:
Turning it back to `Stable.V1.t`

This is why we should remove as many functor parameters as possible. This removes the assumption that if a functor module parameter is stable, then only one unique module is in place of that module. It's quite redundant.

Reply:
Indeed!